packages/flutter/lib/src/material/switch.dart

```diff
@@ -915,18 +915,23 @@ class _SwitchPainter extends ToggleablePainter {
     // Paint the track
     final Paint paint = Paint()
       ..color = trackColor;
+
     const double trackHorizontalPadding = kRadialReactionRadius - _kTrackRadius;
+    const double trackWidth = _kSwitchWidth - 2.0 * trackHorizontalPadding;
+    const double radiusAdditionalOffset = _kTrackRadius / 2;
+    final double parentWidthOffset = size.width / 2 - trackWidth + radiusAdditionalOffset;
```

Comment:
I am not fully following along how the parentWidthOffset has been determined, can you explain that?

I thought it would just be `(size.width - _kSwitchWidth) / 2.0`.

Reply:
I've refactored the implementation a bit to make it easier to follow, hope that helps

Reply:
This makes it .. more confusing to follow along. I think the previous implementation was easier to understand. Can you just give a verbal explanation of why the `parentWidthOffset` has to be `size.width / 2 - trackWidth + radiusAdditionalOffset` over just `(size.width - _kSwitchWidth) / 2.0`.

Reply:
do you mind pointing out what's more confusing? 

Also, while refactoring I've found that my previous implementation was wrong as I was confused due to the amount of informations. I read the material design switch spec which made things clearer and so I believe the current solution to be correct.

Reply:
@goderbauer perhaps I could add some comments? I think it's really hard to assign meaningful names to the variables I'm modifying, 1-2 comment lines per logic block would probably explain a lot

Reply:
Some comments would probably help.

Reply:
@goderbauer do you mind having a look? I hope it's better now